conn.go

```diff
@@ -73,13 +73,20 @@ type Conn struct {
 	requiredAcks int32
 	apiVersions  map[apiKey]ApiVersion
 	fetchVersion apiVersion
+
+	transactionalId *string
 }
 
 // ConnConfig is a configuration object used to create new instances of Conn.
 type ConnConfig struct {
 	ClientID  string
 	Topic     string
 	Partition int
+
+	// The transactional id to use for transactional delivery. Idempotent
+	// deliver should be enabled if transactional id is configured.
+	// For more details look at transactional.id description here: http://kafka.apache.org/documentation.html#producerconfigs
+	TransactionalId *string
```

Comment:
This pointer-to-string model is not used anywhere else in kafka-go, it seems a bit out of place to introduce this.

Could we use the zero-value as a way to express that there is no transactional ID? Or is an empty string valid in this case?

Reply:
From the description of transactional.id I infer that empty string is a valid value: http://kafka.apache.org/documentation.html#producerconfigs
TransactionalId is a nullable string: https://kafka.apache.org/protocol#protocol_types
We need to be able to distinguish between empty value and no value. It doesn't need to be a pointer, but then we need to decide how to communicate no value. We can probably use something like Optional(Maybe), but then it will be even more against the current design and not idiomatic to go.

Reply:
<img width="991" alt="Screen Shot 2019-03-17 at 12 38 27 PM" src="https://user-images.githubusercontent.com/865510/54496900-9a7ff880-48b1-11e9-8843-82788c72ed57.png">

The forth column describes "valid values", seems like even if it's represented as a nullable string on the wire it cannot be an empty string?

Reply:
Indeed. I didn't pay attention to this column. I'll use empty string as default value.